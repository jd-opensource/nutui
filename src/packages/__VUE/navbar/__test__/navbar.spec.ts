import { mount, config } from '@vue/test-utils';
import NavBar from '../index.vue';

import NutIcon from '../../icon/index.vue';

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('should render left slot correctly', () => {
  const wrapper = mount(NavBar, {
    slots: {
      left: () => 'Custom Left'
    }
  });

  expect(wrapper.find('.nut-navbar__left').html()).toMatchSnapshot();
});

test('should render left slot correctly', () => {
  const wrapper = mount(NavBar, {
    slots: {
      right: () => 'Custom Right'
    }
  });

  expect(wrapper.find('.nut-navbar__right').html()).toMatchSnapshot();
});

test('should render title slot correctly', () => {
  const wrapper = mount(NavBar, {
    slots: {
      title: () => 'Custom Title'
    }
  });

  expect(wrapper.find('.nut-navbar__title').html()).toMatchSnapshot();
});

test('should emit click-right event when clicking right text', () => {
  const wrapper = mount(NavBar, {
    props: {
      desc: 'right'
    }
  });

  wrapper.find('.right_text').trigger('click');
  expect(wrapper.emitted('on-click-right')).toBeTruthy();
});
