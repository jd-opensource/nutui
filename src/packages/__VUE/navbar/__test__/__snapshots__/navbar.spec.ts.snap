// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render left slot correctly 1`] = `"<view class=\\"nut-navbar__left\\"><i class=\\"nutui-iconfont nut-icon nut-icon-left\\" style=\\"color: rgb(151, 151, 151);\\" src=\\"\\"></i>Custom Left</view>"`;

exports[`should render left slot correctly 2`] = `
"<view class=\\"nut-navbar__right\\">
  <!--v-if-->Custom Right
</view>"
`;

exports[`should render title slot correctly 1`] = `
"<view class=\\"nut-navbar__title\\">
  <!--v-if-->
  <!--v-if-->
</view>"
`;
